### PR TITLE
Timur katimullin 11 20 tasks

### PR DESCRIPTION
**Pull request by linkouth**
Adds solutions 11-20 tasks

### CODE REVIEW

File: tasks/14_clone-arr/clone-arr.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+/**
+ * cloneArr - returns array deep clone
+ *
+ * @param {Array<*>} arr
+ * @return {Array<*>}
+ */
+export function cloneArr(arr) {
+  let clonedArray = [];
```
Comment (alexkond) [inline tasks/14_clone-arr/clone-arr.js]: This solution isn't optimal. This could be done without `forEach()` and extra array.

File: tasks/16_repeat-string/repeat-string.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
+/**
+ * repeatStringNumTimes - repeats a given string num times
+ *
+ * @param {String} str
+ * @param {Number} num
+ * @return {String}
+ */
+export function repeatStringNumTimes(str, num) {
+  let result = '';
+  for (let i = 0; i < num; i++) {
```
Comment (alexkond) [inline tasks/16_repeat-string/repeat-string.js]: Look closer to `String` prototype, it has method for this case.

File: tasks/18_humanize-format/humanize-format.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+ * @return {String}
+ */
+export function humanizeFormat(number) {
+  if (number === undefined) {
+    return '';
+  }
+  let stringNumber = number.toString();
+  let lastChar = stringNumber.substring(
+      stringNumber.length - 1, stringNumber.length);
+  if (lastChar === '1') {
```
Comment (alexkond) [inline tasks/18_humanize-format/humanize-format.js]: I think you could use `switch` in this case.

**Review by alexkond — CHANGES_REQUESTED** [review]

File: tasks/11_find-longest-word/find-longest-word.js [hunk 4]
```diff
@@ -0,0 +1,14 @@
+/**
+ * findLongestWord - returns the length of the longest
+ * word in the provided sentence
+ *
+ * @param {String} str
+ * @return {Number}
+ */
+export function findLongestWord(str) {
+  let words = str.split(' ');
```
Comment (vovcyan) [inline tasks/11_find-longest-word/find-longest-word.js]: const

File: tasks/11_find-longest-word/find-longest-word.js [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+  let getMax = (firstArg, secondArg) =>
```
Comment (vovcyan) [inline tasks/11_find-longest-word/find-longest-word.js]: Please remove function expression here
Lets put anon function into map method

File: tasks/11_find-longest-word/find-longest-word.js [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+    firstArg.length > secondArg.length ? firstArg : secondArg;
+  let maxLengthWord = words.reduce(getMax, 0);
```
Comment (vovcyan) [inline tasks/11_find-longest-word/find-longest-word.js]: const

File: tasks/11_find-longest-word/find-longest-word.spec.js [hunk 7]
```diff
@@ -1,6 +1,6 @@
-// import { findLongestWord } from './find-longest-word';
```
Comment (vovcyan) [inline tasks/11_find-longest-word/find-longest-word.spec.js]: Please revert this change

File: tasks/12_title-case/title-case.js [hunk 8]
```diff
@@ -0,0 +1,12 @@
+/**
+ * titleCase - Return the provided string
+ * with the first letter of each word capitalized
+ *
+ * @param {String} str
+ * @return {String}
+ */
+export function titleCase(str) {
+  let words = str.toLowerCase().split(' ');
```
Comment (vovcyan) [inline tasks/12_title-case/title-case.js:9]: const

File: tasks/12_title-case/title-case.js [hunk 9]
```diff
@@ -0,0 +1,12 @@
...
+  words = words.map((word) => word.charAt(0).toUpperCase() + word.substr(1));
```
Comment (vovcyan) [inline tasks/12_title-case/title-case.js]: Please combine line 10 and 11

File: tasks/12_title-case/title-case.spec.js [hunk 10]
```diff
@@ -1,6 +1,6 @@
-// import { titleCase } from './title-case';
```
Comment (vovcyan) [inline tasks/12_title-case/title-case.spec.js]: Please revert it

**Review by vovcyan — CHANGES_REQUESTED** [review]

File: tasks/14_clone-arr/clone-arr.js (hunk 1 above)
Comment (linkouth) [reply tasks/14_clone-arr/clone-arr.js]: Fixed

File: tasks/18_humanize-format/humanize-format.js (hunk 3 above)
Comment (linkouth) [reply tasks/18_humanize-format/humanize-format.js]: Fixed

File: tasks/16_repeat-string/repeat-string.js (hunk 2 above)
Comment (linkouth) [reply tasks/16_repeat-string/repeat-string.js]: Fixed

File: tasks/11_find-longest-word/find-longest-word.js (hunk 4 above)
Comment (linkouth) [reply tasks/11_find-longest-word/find-longest-word.js]: Fixed

File: tasks/11_find-longest-word/find-longest-word.js (hunk 5 above)
Comment (linkouth) [reply tasks/11_find-longest-word/find-longest-word.js]: Fixed

File: tasks/11_find-longest-word/find-longest-word.js (hunk 6 above)
Comment (linkouth) [reply tasks/11_find-longest-word/find-longest-word.js]: Fixed

File: tasks/11_find-longest-word/find-longest-word.spec.js (hunk 7 above)
Comment (linkouth) [reply tasks/11_find-longest-word/find-longest-word.spec.js]: Reverted

File: tasks/12_title-case/title-case.js (hunk 8 above)
Comment (linkouth) [reply tasks/12_title-case/title-case.js:9]: Fixed

File: tasks/12_title-case/title-case.js (hunk 9 above)
Comment (linkouth) [reply tasks/12_title-case/title-case.js]: Fixed

File: tasks/12_title-case/title-case.spec.js (hunk 10 above)
Comment (linkouth) [reply tasks/12_title-case/title-case.spec.js]: Reverted

**Review by vovcyan — DISMISSED** [review]

File: tasks/12_title-case/title-case.js (hunk 8 above)
Comment (alexkond) [reply tasks/12_title-case/title-case.js:9]: @linkouth Also please press "Resolve conversation" button when you fixed change request.

**Review by alexkond — DISMISSED** [review]

File: tasks/14_clone-arr/clone-arr.js [hunk 11]
```diff
@@ -0,0 +1,9 @@
+/**
+ * cloneArr - returns array deep clone
```
Comment (vadeneev) [inline tasks/14_clone-arr/clone-arr.js:2]: Array.from makes shallow copy - not deep

**Review by vadeneev — DISMISSED** [review]

File: tasks/14_clone-arr/clone-arr.js (hunk 11 above)
Comment (linkouth) [reply tasks/14_clone-arr/clone-arr.js:2]: Fixed

**Review by alexkond — DISMISSED** [review]

File: tasks/14_clone-arr/clone-arr.js (hunk 11 above)
Comment (vadeneev) [reply tasks/14_clone-arr/clone-arr.js:2]: Spread same, i mean - you have wrong method's description

**Review by vadeneev — APPROVED** [review]

**Review by vovcyan — APPROVED** [review]

**Review by KalachevDev — APPROVED** [review]
